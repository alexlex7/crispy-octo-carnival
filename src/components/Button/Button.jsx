import React from 'react';
import styles from './Button.module.css';
import PropTypes from 'prop-types';
import { ReactComponent as SearchIcon } from '../../assets/images/searchIcon.svg';
import { ReactComponent as MenuIcon } from '../../assets/images/menu.svg';
import { ReactComponent as MobileMenuIcon } from '../../assets/images/mobileMenuIcon.svg';

export default function Button({ type, onClick }) {
  return (
    <button className={styles.btn} type="button" onClick={onClick}>
      {type === 'search' && <SearchIcon className={styles.icon} />}
      {type === 'menu' && <MenuIcon className={styles.icon} />}
      {type === 'mobile menu' && <MobileMenuIcon className={styles.icon} />}
    </button>
  );
}

Button.propTypes = {
  type: PropTypes.oneOf(['search', 'menu', 'mobile menu']).isRequired,
  onClick: PropTypes.func,
};
