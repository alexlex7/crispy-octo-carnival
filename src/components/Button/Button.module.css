.btn {
  padding: 6px;

  border: none;
  background-color: transparent;
  cursor: pointer;
  border-radius: 5px;
  transition: all 0.5s ease;
}

.btn:hover {
  background-color: #244d4d;
}

.icon {
  display: block;

  fill: #244d4d;
  transition: all 0.5s ease;
}
.btn:hover > .icon {
  fill: white;
}
