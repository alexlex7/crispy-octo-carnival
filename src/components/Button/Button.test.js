import { render, screen } from '@testing-library/react';
import Button from './Button';

test('renders search button component', () => {
  const { asFragment } = render(<Button type="search" />);
  const btn = screen.getByRole('button');
  expect(btn).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('renders menu button component', () => {
  const { asFragment } = render(<Button type="menu" />);
  const btn = screen.getByRole('button');
  expect(btn).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('renders mobile menu button component', () => {
  const { asFragment } = render(<Button type="mobile menu" />);
  const btn = screen.getByRole('button');
  expect(btn).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
