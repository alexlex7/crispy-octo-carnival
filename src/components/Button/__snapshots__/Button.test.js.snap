// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders menu button component 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    <svg
      class="icon"
    >
      menu.svg
    </svg>
  </button>
</DocumentFragment>
`;

exports[`renders mobile menu button component 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    <svg
      class="icon"
    >
      mobileMenuIcon.svg
    </svg>
  </button>
</DocumentFragment>
`;

exports[`renders search button component 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    <svg
      class="icon"
    >
      searchIcon.svg
    </svg>
  </button>
</DocumentFragment>
`;
